src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -149,9 +150,16 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
             addr = self.mgr.inventory.get_addr(dd.hostname)
             dashboard_urls.append(build_url(scheme=proto, host=addr, port=port))
 
+        for dd in self.mgr.cache.get_daemons_by_service('snmp-gateway'):
+            assert dd.hostname is not None
+            assert dd.ports
+            deps.append(dd.name())
+            snmp_gateway_urls.append(f"http://{dd.hostname}:{dd.ports[0]}/alerts")
```

Comment:
won't work for ipv6

Reply:
ACK, now using dd.ip or the addr from the host

Reply:
this won't work for clusters without DNS. I mean, we already have 4 examples of this loop already. I think it's time to refactor this now. Let's avoid introducing regressions by adding yet another copy of this loop.